monitoring/uss_qualifier/scd/data_interfaces.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional, List, Dict
+from monitoring.monitorlib.typing import ImplicitDict
+from monitoring.monitorlib.scd_automated_testing.scd_injection_api import InjectFlightRequest
+from monitoring.uss_qualifier.common_data_definitions import Severity
+
+### Begin Definitions around users and test executor 
+
+class Result(ImplicitDict):
+    ''' A class to hold the expected output from data processing and the test interface of the USS under test '''
+    processing_outcome: str
+        
+class KnownIssueFields(ImplicitDict):
+    ''' A class to hold a message that the test executor can provide to the USS in cases when the USS provides a response that is not same as the expected result. '''    
+    test_code: str
+    '''Code corresponding to check generating this issue'''
+
+    relevant_requirements: List[str] = []
+    '''Requirements that this issue relates to'''
+
+    severity: Severity
+    '''How severe the issue is'''
+
+    subject: Optional[str]
+    ''' Identifier of the subject of this issue, if applicable. This may be a UAS serial number, or any field of other object central to the issue. '''
+
+    summary: str
+    '''Human-readable summary of the issue'''
+
+    details: str
+    '''Human-readable description of the issue'''
+
+class RequiredResult(ImplicitDict):
+    ''' A class to evaluate results / response to an injection of test flight data (TestFlightRequest) '''
+    acceptable_results: List[str] # Holds information about the "result" response provided by the test interface of the USS under test. Per the SCD Testing API this should be one of "Planned", "Rejected", "ConflictWithFlight" or "Failed"
+    incorrect_result_details: Optional[Dict[Result, KnownIssueFields]] # In the case where the USS under test provides a result that is not the same as expected result then the test driver can provide a message detailing why the expected result was populated the way it was. This should help the USS under test debug specific parts of their internal systems. 
+
+### End of definitions aronud users and test executor
+
+### Definitions around interface between scd data generation and test executor 
+  
+class InjectionTarget(ImplicitDict):
+    ''' A class to hold details of the USS under test '''
+    uss_sequence_number: int
+    name: str
+
+class FlightInjectionAttempt(ImplicitDict):
+    ''' A class to hold details of the test injection, the injection target and the expected result of processing the flight request '''
+    test_injection: InjectFlightRequest
+    required_result: RequiredResult
+    injection_target: InjectionTarget
+
+class TestDetails(ImplicitDict):
+    ''' A class to hold details of different test structures '''
+    name: str # e.g. a Nominal test or Flight Authorisation test
+    response_wait_time_secs: int # amount of time the test driver should wait before expecting a response
```

Comment:
The `FlightInjectionAttempt` class contains nearly all information about an attempt to inject a flight into a USS: it contains a description of the flight that should be injected, to which USS that flight should be injected, and what test executor should do based on the response it receives from that injection.  Why would we put one aspect of that flight injection attempt in a separate class like this?  Instead, it should be part of `FlightInjectionAttempt`.  The place where would probably make most sense would probably be in the class currently named `RequiredResult` (to be renamed; see other comment) since that class contains information about what test executor should do if it gets various responses, and no response is a response in that context.  The test executor should also be told what to do if it hits this timeout (and thus receives no response) -- the way we're accomplishing that for other types of responses is via a `KnownIssueFields` instance.

Reply:
Thank you I have added a new timeout response field as a `KnownIssueFields` instance.